cmd/thanos/store.go

```diff
@@ -353,7 +353,7 @@ func runStore(
 
 	options := []store.BucketStoreOption{
 		store.WithLogger(logger),
-		store.WithRegistry(reg),
```

Comment:
Why? Metric name should mean component, not necessarily the target which sources those. This gives amazing characteristics of same dashboards/alerts working for ANYHING that uses objstore. See second bullet for https://prometheus.io/docs/practices/naming/#metric-names

Reply:
Unless we want to avoid painful breaking change...

Reply:
I just merged https://github.com/thanos-io/objstore/pull/26. 
Given we moved that component out of Thanos itself, it should be fine to stand alone as `objstore_` metrics. 
Now the question is if Thanos wants to add the prefix `thanos_` back to not break everyone, or move to more generic `objstore_` metrics only. 
Then alerts with those metrics would work for example for Parca too.

Reply:
@bwplotka you tell us 😄 

As someone using Thanos, together with the dashboards and mixins from kube-thanos, I prefer if there aren't breaking changes. But if maintainers decide the path forward is as @metalmatze suggests (dropping the `thanos_` prefix for Thanos Store metrics and keeping only the generic `objstore_` one), we can go on with it.

Reply:
I'd also prefer to keep the `thanos_` prefix. Also `objstore_` seems too generic as a component identifier.

Reply:
What's wrong with objstore? It's all metrics about buckets operations, so semantically it makes sense, no matter who performs those bucket operations (job or any other target label will tell you). Again it allows `objstore_` ALERTs and dashboards to be state of the art, reusable by anybody using objstore component.

But as per compatibility breaking change I will let you decide, since I won't have time to fix dashboards and alerts. I know many Thanos users have custom alerts on this, so it might be painful. (: 

Perhaps wrap it for now (don't break) and add issue to do some bigger metric overhaul/braking change later so it's one specific release where everybody has to adjust alerts/dashboards at once?

We could think about deprecation mode, where we duplicate metric or put it behind compatibility flag as well 🤔

Reply:
FYI See official guide: https://prometheus.io/docs/practices/naming/#metric-names

```
...should have a (single-word) application prefix relevant to the domain the metric belongs to. The prefix is sometimes referred to as namespace by client libraries. For metrics specific to an application, the prefix is usually the application name itself. Sometimes, however, metrics are more generic, like standardized metrics exported by client libraries. Examples:
prometheus_notifications_total (specific to the Prometheus server)
process_cpu_seconds_total (exported by many client libraries)
http_request_duration_seconds (for all HTTP requests)
```

Reply:
I think this is a super great idea that we should keep in mind but for now let's stay focused in this PR :+1